### PR TITLE
Fix #3324: Create a BUILD.bazel file under domain model in oppialogger package

### LINKED ISSUE #3324: Create a BUILD.bazel file for domain/oppialogger
This task is part of the build system migration from Gradle to Bazel. 

## Prerequesites

- Make sure you are able to build Oppia using Bazel
- Follow the [Oppia Bazel Setup instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions) to setup Bazel and learn the terminology.
- Read the [Gradle Bazel Migration Best Practies and FAQ](https://github.com/oppia/oppia-android/wiki/Gradle--Bazel-Migration-Best-Practices-and-FAQ) for more context and see if there's an answer there to any question you have.

## Task

1. Create a BUILD.bazel file in

`domain/src/main/java/org/oppia/android/domain/oppialogger`

and its subdirectories.

1. Define the kt_android_library library/libraries

Based on the imports in the Kotlin files, add the required third_party dependencies to make the libraries build. Check the Useful links section for a list of third_party dependencies.  You may also need to add the non third_party dependencies.

Main consideration here is the dependencies of the subpackages.

- loguploader depends on exceptions, analytics, ApplicationStartupListener.kt
- exceptions depends on LogStorageModule.kt
- analytics depends on LogStorageModule.kt

3. Add the directory to the MIGRATED_PROD_FILES of the module's root BUILD.bazel file

4. Add the libraries to the list of deps dependencies in the targets that require the library

You will have to set the apropriate visbility. See the list of visibilities in the link below.

5. Clean up any unused deps dependencies

6. Verify that the app builds

## Useful links

- [BUILD.bazel files](https://github.com/oppia/oppia-android/search?q=filename%3ABUILD.bazel&type=code)
- [Visibilities](https://github.com/oppia/oppia-android/blob/76dc13dba3ad77aecd2218a5372949d6551f12fe/BUILD.bazel)
- [third_party dependencies](https://github.com/oppia/oppia-android/blob/a70d9b016f0d77dd30c4939041f82ea02ee28d1d/third_party/versions.bzl)

### PR DESCRIPTION
**Pull request by jonathanalvares9009**
## Explanation
Created a BUILD.bazel file under domain model in oppialogger package.
Fixes #3324 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by jonathanalvares9009** [comment]
@fsharpasharp could you give an initial review? I'll fix the failing tests

**Review by fsharpasharp — COMMENTED** [review]
Thanks, nice that you're working with the visibilities. My main suggestion is to rename the libraries to shorter names. For example, uncaught_exception_logger_module could be prod_module or module, since the libraries are also qualified by their paths.
Another example is that exceptions_controller could just be controller. This comment applies more to the targets with broader visibility. I'll defer to @BenHenning on more specifics when it comes to naming the libraries.

The difference coming from the util module, is that we're now thinking in packages while some of util were individual libraries.

**Comment by jonathanalvares9009** [comment]
Ok got it, will make the changes

**Review by fsharpasharp — APPROVED** [review]
Looks good to me. I'm assigning @BenHenning for a final pass since he might have some considerations concerning the structure/naming.

**Comment by jonathanalvares9009** [comment]
@fsharpasharp, I have assigned myself bazel issues which were open should I start working on them or wait until these two get merged.

**Comment by fsharpasharp** [comment]
You can work on them, they should be fairly independent. Maybe leave the ones with the good first issue tag.

**Comment by jonathanalvares9009** [comment]
Yeah didn't take those up

**Comment by jonathanalvares9009** [comment]
@anandwana001 could you re-run the tests as they pass locally

**Comment by anandwana001** [comment]
I think the error is flaky, you can ignore it.

**Comment by BenHenning** [comment]
Hi @jonathanalvares9009. Apologies for not looking at this earlier--I plan to review this tomorrow.

**Comment by jonathanalvares9009** [comment]
Hi @BenHenning, no problem

**Comment by BenHenning** [comment]
Sorry @jonathanalvares9009 I'm a bit limited in my reviewer time at the moment & a bunch of folks have deadlines this week. I'll try to get to this tomorrow my time--super sorry about the repeated delays here & for #3261.

**Comment by jonathanalvares9009** [comment]
@BenHenning I understand no problem

File: domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel
Comment (BenHenning) [inline domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel:2]:
```suggestion
Library for providing logging to the Oppia Android app.
```

Ditto in other files.

File: domain/BUILD.bazel
Comment (BenHenning) [inline domain/BUILD.bazel:16]: Aren't these files also still migrated? We should probably keep them in the list.

File: domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/BUILD.bazel
Comment (BenHenning) [inline domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/BUILD.bazel:12]: Does  it need to be ``__subpackages__`` or can it just be ``__pkg__``?

File: domain/src/main/java/org/oppia/android/domain/oppialogger/exceptions/BUILD.bazel
Comment (BenHenning) [inline domain/src/main/java/org/oppia/android/domain/oppialogger/exceptions/BUILD.bazel:44]: For this & controller: could this be ``__pkg__``, instead? In general, prefer to use \_\_pkg\_\_ when possible because it will nicely break when we migrate things, forcing us to use a more restrictive visibility (which is generally preferred since it reduces components inadvertently being used in places that they aren't mean to be used.

Ditto elsewhere in the PR.

File: domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel
Comment (BenHenning) [inline domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel:34]: Why this visibility?

File: domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel
Comment (BenHenning) [inline domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel:22]: This is actually a bit odd to be in this package. Could you maybe move ApplicationStartupListener to be in a new utility/startup package? It will be a small refactor, but I think it's a better place for the startup listener definition to live.

Ditto for ApplicationStartupListenerModule--that ought to be in utility/startup as well.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @jonathanalvares9009! This overall looks quite good. Just had a few comments on visibility & placement of things--please take a look.

Also, could you update this PR to the latest develop? Some new CI checks are missing from below.

**Comment by jonathanalvares9009** [comment]
Yeah sure @BenHenning

File: domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel
Comment (jonathanalvares9009) [reply domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel:2]: Done

File: domain/BUILD.bazel
Comment (jonathanalvares9009) [reply domain/BUILD.bazel:16]: Sorry didn't realise this thanks for bring it to my notice @BenHenning

File: domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/BUILD.bazel
Comment (jonathanalvares9009) [reply domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/BUILD.bazel:12]: I feel this should remain  __subpackages__ as it is used only inside oppialogger module. Any suggestions @BenHenning

File: domain/src/main/java/org/oppia/android/domain/oppialogger/exceptions/BUILD.bazel
Comment (jonathanalvares9009) [reply domain/src/main/java/org/oppia/android/domain/oppialogger/exceptions/BUILD.bazel:44]: Done

File: domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel
Comment (jonathanalvares9009) [reply domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel:34]: @BenHenning used this visibility because it is only used in the oppialogger module. Any suggestions.

File: domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel
Comment (jonathanalvares9009) [reply domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel:22]: Hi @BenHenning, you mean move the ApplicationStartupListenerModule from app into utility/startup

File: domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel
Comment (BenHenning) [reply domain/src/main/java/org/oppia/android/domain/oppialogger/BUILD.bazel:34]: Could we set it to the specific __pkg__ if it's just one subpackage that depends on this? Or, to private if it's only a class in the same package?

File: domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/BUILD.bazel
Comment (BenHenning) [reply domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/BUILD.bazel:12]: We probably shouldn't be thinking of 'oppialogger/' as "oppialogger/ + its children" but rather as a independent package to the child packages. From that perspective, we shouldn't over-expose any libraries beyond the minimum packages that need to depend on them unless they are part of the testing or public API of the entire layer (in which cases we use the broad visibility package groups).

**Comment by BenHenning** [comment]
Thanks @jonathanalvares9009! Followed up to your comments--PTAL. Also, you mentioned 'Done' to some of the comments but the changes haven't yet been uploaded--is that intentional?

**Comment by jonathanalvares9009** [comment]
> Thanks @jonathanalvares9009! Followed up to your comments--PTAL. Also, you mentioned 'Done' to some of the comments but the changes haven't yet been uploaded--is that intentional?

I mentioned done and did it on local but didn't push it sorry.

**Pull request closed without merging**
